test/test_api/test_api.py

```diff
@@ -405,29 +406,31 @@ def test_tabular_regression(openml_name, resampling_strategy, backend, resamplin
     assert 'Estimator' in representation
 
 
-@unittest.mock.patch('autoPyTorch.evaluation.train_evaluator.eval_train_function', new=dummy_eval_train_function)
```

Comment:
why do you not want to use the patch, without it, it will take a long time to finish this test and we already test the time series pipeline [here](https://github.com/dengdifan/Auto-PyTorch/blob/time_series_forecasting/test/test_pipeline/test_time_series_forecasting_pipeline.py).

Reply:
unittest.mock.patch does not work for the function wrapped by `partial`. Currently, I cannot find a very good solution for that

Reply:
that is another reason why we should have them as separate functions, one for time series tasks and the other for tabular tasks. Also, I have noticed that you have a failed test for tabular classification API with cross-validation. I suspect it's not due to your PR, but it's my old error. Actually, once we changed the import statement from [`import autoPyTorch.evaluation.train_evaluator](https://github.com/automl/Auto-PyTorch/blob/c0fb82ed8ae3d29cd90f72b0e86ff9bf499140f8/autoPyTorch/evaluation/tae.py#L25) to [`from autoPyTorch.evaluation.train_evaluator import eval_train_function`](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/autoPyTorch/evaluation/tae.py#L38) the patch stopped working. If you want I can make a small PR to fix this issue. But it may require you to rebase your branch, could you make the changes. 

You only have to change the patch line [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L47), [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L224), [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L673) and [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L715) from `@unittest.mock.patch('autoPyTorch.evaluation.train_evaluator.eval_train_function', new=dummy_eval_train_function)` to `@unittest.mock.patch('autoPyTorch.evaluation.tae.eval_train_function', new=dummy_eval_train_function)` since `eval_train_function` is now in `tae`'s namespace. 

Let me know what you think.

Reply:
Also, I have noticed that you have a failed test for tabular classification API with cross-validation. I suspect it's not due to your PR, but it's my old error. Actually, once we changed the import statement from [`import autoPyTorch.evaluation.train_evaluator](https://github.com/automl/Auto-PyTorch/blob/c0fb82ed8ae3d29cd90f72b0e86ff9bf499140f8/autoPyTorch/evaluation/tae.py#L25) to [`from autoPyTorch.evaluation.train_evaluator import eval_train_function`](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/autoPyTorch/evaluation/tae.py#L38) the patch stopped working. If you want I can make a small PR to fix this issue. But it may require you to rebase your branch, could you make the changes. 

You only have to change the patch line [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L47), [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L224), [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L673) and [here](https://github.com/automl/Auto-PyTorch/blob/bb7f5c5648c16f61cba7d412de30546650060ce3/test/test_api/test_api.py#L715) from `@unittest.mock.patch('autoPyTorch.evaluation.train_evaluator.eval_train_function', new=dummy_eval_train_function)` to `@unittest.mock.patch('autoPyTorch.evaluation.tae.eval_train_function', new=dummy_eval_train_function)` since `eval_train_function` is now in `tae`'s namespace. 

Let me know what you think.

Reply:
https://github.com/automl/Auto-PyTorch/pull/434/commits/acddd2284fc783589488dfbe7011212eb92e9142

Reply:
Thanks